src/pybind/mgr/mgr_util.py

```diff
@@ -409,6 +413,23 @@ def format_bytes(n: int, width: int, colored: bool = False) -> str:
     return format_units(n, width, colored, decimal=False)
 
 
+def test_port_allocation(addr: str, port: int) -> bool:
+    """Checks if the port is bound on addr
+    :raises PortAlreadyInUse: in case port is already in use
+    :raises Exception: any generic error other than port already in use
+    """
```

Comment:
```suggestion
def check_port_in_use(addr: str, port: int) -> bool:
    """Checks if the port is available
    :raises PortAlreadyInUse: in case port is already in use
    :raises Exception: any generic error other than port already in use
    :return False if port is available
    """
```
NIT on naming and comment, feel free to take/leave whatever parts you agree/disagree with

Reply:
I removed the `return bool` because right now we rely on the `PortAlreadyInUse` exception to test if port is available or not.